plugins/terraform/layer0/resource_layer0_service_test.go

```diff
@@ -66,6 +69,68 @@ func TestResourceServiceCreateRead(t *testing.T) {
 	assert.Equal(t, "lb_id", d.Get("load_balancer").(string))
 	assert.Equal(t, "dpl_id", d.Get("deploy").(string))
 	assert.Equal(t, 3, d.Get("scale").(int))
+	assert.Equal(t, false, d.Get("stateful").(bool))
+}
+
+func TestResourceServiceCreateRead_stateful(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockClient := mock_client.NewMockClient(ctrl)
+
+	defer client.SetTimeMultiplier(0)()
+
+	req := models.CreateServiceRequest{
+		ServiceName:    "svc_name",
+		EnvironmentID:  "env_id",
+		LoadBalancerID: "lb_id",
+		DeployID:       "dpl_id",
+		Scale:          3,
+		ServiceType:    models.DeployCompatibilityStateful,
+	}
+
+	mockClient.EXPECT().
+		CreateService(req).
+		Return("svc_id", nil)
+
+	service := &models.Service{
+		ServiceID:      "svc_id",
+		ServiceName:    "svc_name",
+		ServiceType:    models.DeployCompatibilityStateful,
+		EnvironmentID:  "env_id",
+		LoadBalancerID: "lb_id",
+		DesiredCount:   3,
+		Deployments: []models.Deployment{
+			{DeployID: "dpl_id", Status: "PRIMARY", DesiredCount: 1, RunningCount: 1},
+		},
+	}
+
+	mockClient.EXPECT().
+		ReadService("svc_id").
+		Return(service, nil).
+		AnyTimes()
+
+	serviceResource := Provider().(*schema.Provider).ResourcesMap["layer0_service"]
+	d := schema.TestResourceDataRaw(t, serviceResource.Schema, map[string]interface{}{
+		"name":          "svc_name",
+		"environment":   "env_id",
+		"load_balancer": "lb_id",
+		"deploy":        "dpl_id",
+		"scale":         3,
+		"stateful":      true,
```

Comment:
it seems a little odd that we mix bools and strings for the same field. perhaps instead of `ServiceType/TaskType string` fields in the models, they should just have `Stateful bool` fields?

Reply:
You're right - takin' this train to Boolville. ✅